src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,65 @@
+package duke;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.data.Storage;
+import duke.data.TaskList;
+import duke.dukeexceptions.DukeException;
+import duke.parser.Parser;
+import duke.responses.Response;
+import duke.ui.Ui;
+
+/***
+ * Duke is a program that takes in user inputs and stores them as Tasks
+ */
+public class Duke {
+    private TaskList taskList;
+    private Storage store;
+    private Parser commandHandler;
+    private Ui cmdLine;
+    private boolean isRunning = true;
+
+    /**
+     * Constructor for Duke Chatbot
+     */
+    public Duke() {
+        this.taskList = new TaskList();
+        this.store = new Storage(taskList);
+        this.commandHandler = new Parser();
+        this.cmdLine = new Ui();
```

Comment:
Would it be less redundant if `this` keyword was not used?
```suggestion
        taskList = new TaskList();
        store = new Storage(taskList);
        commandHandler = new Parser();
        cmdLine = new Ui();
```